web-common/src/runtime-client/watch-request-client.ts

```diff
@@ -0,0 +1,82 @@
+import type {
+  V1WatchFilesResponse,
+  V1WatchLogsResponse,
+  V1WatchResourcesResponse,
+} from "@rilldata/web-common/runtime-client/index";
+import { streamingFetchWrapper } from "@rilldata/web-common/runtime-client/fetch-streaming-wrapper";
+import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import type { Runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+import { asyncWait } from "@rilldata/web-local/lib/util/waitUtils";
+import { get, Unsubscriber } from "svelte/store";
+
+type WatchResponse =
+  | V1WatchFilesResponse
+  | V1WatchResourcesResponse
+  | V1WatchLogsResponse;
+
+type StreamingFetchResponse<Res extends WatchResponse> = {
+  result?: Res;
+  error?: { code: number; message: string };
+};
+
+export class WatchRequestClient<Res extends WatchResponse> {
```

Comment:
Do we need `Res extends WatchResponse` or can we just use `WatchResponse`?

Reply:
This is needed to get typescript styling for the caller.